src/kakarot/instructions/environmental_information.cairo

```diff
@@ -167,63 +112,34 @@ namespace EnvironmentalInformation {
     }(ctx: model.ExecutionContext*) -> model.ExecutionContext* {
         alloc_locals;
 
-        let stack = ctx.stack;
-
-        // Stack input:
-        // 0 - offset: calldata offset of the word we read (32 byte steps).
-        let (stack, calldata_offset) = Stack.pop(stack);
+        let (stack, offset) = Stack.pop(ctx.stack);
 
         let (sliced_calldata: felt*) = alloc();
-
-        let calldata: felt* = ctx.call_context.calldata;
-        let calldata_len: felt = ctx.call_context.calldata_len;
-
-        // read calldata at offset
-        let sliced_calldata: felt* = Helpers.slice_data(
-            data_len=calldata_len, data=calldata, data_offset=calldata_offset.low, slice_len=32
+        slice(
```

Comment:
I wonder why we have to make a copy of the slice of the calldata inside sliced_calldata first,

We could technically accumulate a uint256 by reading the ctx.call_context.calldata starting at offset.low, for 32 indexes?

Though i understand it's not the purpose of this PR

Reply:
I had a big PRs with a lot of changes and was not able to easily understand why I had newly failing test cases, so I've closed it and created a step by step PR, indeed not challenging to much the existing but mainly integrating the new utils. For the record, I've now solved some "maximum offset value exceeded", see https://github.com/kkrt-labs/ef-tests/pull/610.

Then I'll move on to the refacto of the logic